tests/python/TestConnections.py

```diff
@@ -0,0 +1,463 @@
+            }
+        }
+        all_queries.append(fE)
+
+        fE = {
+            "FindEntity": {
+                "class": "felcflo_foo",
+                "constraints": {
+                    "name": ["==", "cat"]
+                },
+                "_ref": 4
+            }
+        }
+        all_queries.append(fE)
+
+        aC = {
+
+            "AddConnection": {
+                "class": "foo_connection",
+                "ref1": 2,
+                "ref2": 3,
+                "properties":{
+                    "name": "best_type_of_connection",
+                    "probablity": 0.3
+                }
+            }
+        }
+        all_queries.append(aC)
+
+        aC = {
+
+            "AddConnection": {
+                "class": "foo_connection",
+                "ref1": 2,
+                "ref2": 4,
+                "properties":{
+                    "name": "best_type_of_connection",
+                    "probablity": 0.6
+                }
+            }
+        }
+        all_queries.append(aC)
+
+        response, res_arr = db.query(all_queries)
+
+        self.assertEqual(response[0]["FindEntity"]["status"], 0)
+        self.assertEqual(response[1]["FindEntity"]["status"], 0)
+        self.assertEqual(response[2]["FindEntity"]["status"], 0)
+        self.assertEqual(response[3]["AddConnection"]["status"], 0)
+        self.assertEqual(response[4]["AddConnection"]["status"], 0)
+
+        all_queries = []
+
+        fE = {
+            "FindEntity": {
+                "class": "felcflo_People",
+                "_ref": 1,
+                "results": {
+                    "list": ["name", "lastname"]
+                }
+            }
+        }
+        all_queries.append(fE)
+
+        fE = {
+            "FindEntity": {
+                "class": "felcflo_foo",
+                "link": {
+                    "ref": 1,
+                    "constraints": {
+                        "probablity": [">=", 0.5],
+                        "name": ["==", "best_type_of_connection"]
+                    }
+
+                },
+                "results": {
+                    "list": ["name"]
+                }
+            }
+        }
+        all_queries.append(fE)
+
+        response, res_arr = db.query(all_queries)
+
+        self.assertEqual(len(response[1]["FindEntity"]["entities"]), 1)
+        self.assertEqual(response[1]["FindEntity"]["entities"][0]["name"], "cat")
+
+        all_queries = []
+
+        fE = {
+            "FindEntity": {
+                "class": "felcflo_People",
+                "_ref": 1,
+                "results": {
+                    "list": ["name", "lastname"]
+                }
+            }
+        }
+        all_queries.append(fE)
+
+        fE = {
+            "FindEntity": {
+                "class": "felcflo_foo",
+                "link": {
+                    "ref": 1,
+                    "constraints": {
+                        "probablity": [">=", 0.1],
```

Comment:
JSON is ugly as it is. I really don't like this nesting. ANd it beats your original point about a simple way to use link.